src/note.rs

```diff
@@ -0,0 +1,394 @@
+        &self.value_commitment
+    }
+
+    /// Attempt to decrypt the note value provided a [`ViewKey`]. Always
+    /// succeeds for transparent notes, and will return random values for
+    /// obfuscated notes provided the wrong view key.
+    pub fn value(&self, vk: Option<&ViewKey>) -> u64 {
+        match self.note_type {
+            NoteType::Transparent => {
+                let bytes = chunk_of!(8; &self.encrypted_data);
+                u64::from_le_bytes(bytes)
+            }
+            NoteType::Obfuscated if vk.is_some() => {
+                let (value, _) = self.decrypt_data(vk.unwrap());
+                value
+            }
+            _ => {
+                let mut rng = rand::thread_rng();
+                let value: u64 = rng.gen();
+                value
+            }
+        }
+    }
+
+    /// Decrypt the blinding factor with the provided [`ViewKey`]
+    ///
+    /// If the decrypt fails, a random value is returned
+    pub fn blinding_factor(&self, vk: Option<&ViewKey>) -> JubJubScalar {
+        match self.note_type {
+            NoteType::Transparent => JubJubScalar::zero(),
+            NoteType::Obfuscated if vk.is_some() => {
+                let (_, blinding_factor) = self.decrypt_data(vk.unwrap());
+                blinding_factor
+            }
+            _ => JubJubScalar::random(&mut rand::thread_rng()),
+        }
+    }
+}
+
+impl Ownable for Note {
+    fn stealth_address<'a>(&'a self) -> &'a StealthAddress {
+        &self.stealth_address
+    }
+}
+
+impl<H: ByteHash> Content<H> for Note {
+    fn persist(&mut self, sink: &mut Sink<H>) -> io::Result<()> {
+        (self.note_type as u8).persist(sink)?;
+
+        sink.write_all(&JubJubAffine::from(&self.value_commitment).to_bytes())?;
+        sink.write_all(&self.stealth_address.to_bytes())?;
+
+        sink.write_all(&self.nonce.to_bytes())?;
+        self.pos.persist(sink)?;
+
+        sink.write_all(&self.encrypted_data)?;
+        Ok(())
+    }
+
+    fn restore(source: &mut Source<H>) -> io::Result<Self> {
+        let mut one_scalar = [0u8; 32];
+        let mut one_stealth_address = [0u8; 64];
+
+        let note_type = u8::restore(source)?.try_into()?;
+
+        source.read_exact(&mut one_scalar)?;
+        let value_commitment =
+            JubJubExtended::from(jubjub_decode::<JubJubAffine>(&one_scalar)?);
+
+        source.read_exact(&mut one_stealth_address)?;
+        let stealth_address = StealthAddress::try_from(&one_stealth_address)?;
+
+        source.read_exact(&mut one_scalar)?;
+        let nonce = jubjub_decode::<JubJubScalar>(&one_scalar)?;
+
+        let pos = u64::restore(source)?;
+
+        let mut encrypted_data = [0u8; ENCRYPTED_DATA_SIZE];
+        source.read_exact(&mut encrypted_data)?;
+
+        Ok(Note {
+            note_type,
+            value_commitment,
+            nonce,
+            stealth_address,
+            pos,
+            encrypted_data,
+        })
+    }
```

Comment:
Shouldn't we add also the `Into<StorageScalar>` trait impl here?

Reply:
I was thinking to add in a follow up PR, in order to land this – since the `StorageScalar` is also part of the `PoseidonTree` functionality that is still missing – but wen can also add the implementation directly in the Initial Commit.
You're the reviewer, let me know what's your preference.

Personally I'm more about progressive enhancement unless is a key function that prevent everyone to contribute to the repo.

Reply:
If you want to do it on a follw-ip PR, feel free to add the issue now.